### PR TITLE
Add single/multiple app deploy using cluster selector

### PR DESCRIPTION
**Pull request by sbulage**
Fixes/Automates: #178

### CODE REVIEW

File: tests/assets/git-repo-multiple-app-cluster-selector.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+spec:
+  branch: master
+  paths:
+    - qa-test-apps/nginx-app
+    - multiple-paths/config
+  repo: https://github.com/rancher/fleet-test-data/
+  targets:
+    - clusterSelector:
+        matchLabels:
+          key_env: value_testing
```
Comment (mmartin24) [inline tests/assets/git-repo-multiple-app-cluster-selector.yaml:15]: pls add exra line if you consider so

File: tests/assets/git-repo-single-app-cluster-selector.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  namespace: fleet-default
+spec:
+  branch: master
+  paths:
+    - qa-test-apps/nginx-app
+  repo: https://github.com/rancher/fleet-test-data/
+  targets:
+    - clusterSelector:
+        matchLabels:
+          key_env: value_testing
```
Comment (mmartin24) [inline tests/assets/git-repo-single-app-cluster-selector.yaml:14]: pls add exra line if you consider so

**Review by mmartin24 — APPROVED** [review]

File: tests/assets/git-repo-multiple-app-cluster-selector.yaml (hunk 1 above)
Comment (sbulage) [reply tests/assets/git-repo-multiple-app-cluster-selector.yaml:15]: Okay, I will add it.

File: tests/assets/git-repo-single-app-cluster-selector.yaml (hunk 2 above)
Comment (sbulage) [reply tests/assets/git-repo-single-app-cluster-selector.yaml:14]: Okay, I will add it.

**Comment by sbulage** [comment]
:green_circle: 2.9 CI Run: https://github.com/rancher/fleet-e2e/actions/runs/10510354503

File: tests/assets/git-repo-single-app-cluster-selector.yaml (hunk 2 above)
Comment (sbulage) [reply tests/assets/git-repo-single-app-cluster-selector.yaml:14]: Unknowingly, I clicked on merge button. Sorry, I have changes made I will push to next immediate PR. :disappointed:

**Pull request merged**